finch.yaml

```diff
@@ -176,7 +176,9 @@ provision:
     mkdir -p  ~/.local/share/cni
     sudo mount --bind /mnt/lima-finch/cni  ~/.local/share/cni
 
-    sudo systemctl restart containerd.service  
+    sudo systemctl restart containerd.service
+    # Restart buildkit to ensure it's using the correct containerd UUID
+    sudo systemctl restart buildkit.service
```

Comment:
Curious if perhaps instead of doing a restart here we should try having `buildkit.service` require `containerd.service`? Thinking out loud there could be other scenarios which lead to containerd restarting which would eventually lead to this behavior.

Reply:
I cannot find the systemd.service files we are using, but it could be done here with the following:

```
sudo systemctl add-requires buildkit.service containerd.service
sudo systemctl restart containerd.service
```

Reply:
Yeah, that works and is more semantically correct and future proof.

Previously I tried `Before` and `After` on buildkit/containerd, but `Requires` is actually what solves it because it actually takes down dependencies.

Our unit files come straight from the `nerdctul-full` archive. The BuildKit unit file is just a simple find/replace of containerd in the containerd unit file, which is probably why it doesn't have anything related to its own dependencies. It should be simple enough to create another sed command that adds the `Requires` line (I need to find where this is done first).

Reply:
Here it is: https://github.com/containerd/nerdctl/blob/29fd529c8a684be58c67c052b4842221542212a7/Dockerfile#L147-L151

Reply:
Something like this should work to add the `Requires=` line after the first new line in the `[Unit]` block:
```
sed -i'' '/^\[Unit\]/!b;:a;n;/./ba;iRequires=containerd.service' /usr/local/lib/systemd/system/buildkit.service
```

Might be better to just replace `[Unit]\n` with `[Unit]\nRequires=containerd.service` though (more readable)

Reply:
Nice, potential upstream contribution.